File: tp-exercices/tp2-classes-passerelle-iot/python/main.py
from capteur import Capteur
from passerelle import Passerelle

class CapteurTemperature(Capteur):
    def acquerir(self) -> bool:
        self._valeur = 23.8
        return True

class CapteurHumidite(Capteur):
    def acquerir(self) -> bool:
        self._valeur = 58.2
        return True

if __name__ == "__main__":
    print("=== DEMO TP 2 : PASSERELLE IOT (PYTHON - BTS CIEL 2) ===")
    passerelle = Passerelle("Gateway-SalleServeur", "192.168.10.254")

    s1 = CapteurTemperature(101, "Sonde Salle Baie 1")
    s2 = CapteurHumidite(102, "Humidite Ambiante")

    passerelle.connecter_capteur(s1)
    passerelle.connecter_capteur(s2)

    passerelle.executer_cycle_mesure()

File: tp-exercices/tp2-classes-passerelle-iot/cpp/main.cpp
#include "Passerelle.hpp"
#include <iostream>

class CapteurTemperature : public Capteur {
public:
    CapteurTemperature(int id, const std::string& label) : Capteur(id, label) {}
    bool acquerir() override {
        m_valeur = 23.8f;
        return true;
    }
};

class CapteurHumidite : public Capteur {
public:
    CapteurHumidite(int id, const std::string& label) : Capteur(id, label) {}
    bool acquerir() override {
        m_valeur = 58.2f;
        return true;
    }
};

int main() {
    std::cout << "=== DEMO TP 2 : PASSERELLE IOT (BTS CIEL 2) ===" << std::endl;

    Passerelle passerelle("Gateway-SalleServeur", "192.168.10.254");

    CapteurTemperature capteurTemp(101, "Sonde Salle Baie 1");
    CapteurHumidite capteurHum(102, "Humidite Ambiante");

    passerelle.connecterCapteur(&capteurTemp);
    passerelle.connecterCapteur(&capteurHum);

    passerelle.executerCycleMesure();

    return 0;
}
